platform/linux_platform.go

```diff
@@ -1169,12 +1169,17 @@ func (p linux) MountPersistentDisk(diskSetting boshsettings.DiskSettings, mountP
 		realPath = partitionPath
 	}
 
-	err = p.diskManager.GetMounter().Mount(realPath, mountPoint, diskSetting.MountOptions...)
-
+	existingFsType, err := p.getPartitionFormatType(partitionPath)
```

Comment:
Instead of duplicating the code from `linuxFormatter#getPartitionFormatType`, could we use the `#IsPersistentDiskMountedExternally` introduced in this same file? Seems more robust for future externally mounted disks.

Reply:
I pushed the changes you suggested. Also the answers to your other question is yes. disk will be mounted by the co-located (luks addon) job.